src/test/java/org/esa/snap/gpt/RunGPTProductReaderTest.java

```diff
@@ -29,7 +34,8 @@
  */
 public class RunGPTProductReaderTest {
 
-    private static final String PROPERTY_NAME_LOG_FILE_PATH = "gpt.tests.log.file";
+    private static final Logger logger = Logger.getLogger(RunGPTProductReaderTest.class.getName());
```

Comment:
Also this class What's it' purpose? GPT test are in the snap-gpt-tests project.
Why have this one here?

Reply:
Moving the class in the module 'gpt-tests-executer' of the 'snap-gpt-tests' products (in the test source folder), add some maven dependencies and running a gp test (for example Jp2ReaderTest.json) the following exception is generated:

java.lang.UnsatisfiedLinkError: no jhdf5 in java.library.path
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1860)
	at java.lang.Runtime.loadLibrary0(Runtime.java:871)
	at java.lang.System.loadLibrary(System.java:1124)
	at ncsa.hdf.hdf5lib.H5.loadH5Lib(H5.java:339)
	at ncsa.hdf.hdf5lib.H5.<clinit>(H5.java:266)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at org.esa.snap.dataio.hdf5.Hdf5ProductWriterPlugIn.loadHdf5Lib(Hdf5ProductWriterPlugIn.java:148)
	at org.esa.snap.dataio.hdf5.Hdf5ProductWriterPlugIn.<clinit>(Hdf5ProductWriterPlugIn.java:45)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at java.util.ServiceLoader$LazyIterator.nextService(ServiceLoader.java:380)
	at java.util.ServiceLoader$LazyIterator.next(ServiceLoader.java:404)
	at java.util.ServiceLoader$1.next(ServiceLoader.java:480)
	at org.esa.snap.core.util.ServiceLoader.loadServices(ServiceLoader.java:40)
	at org.esa.snap.core.dataio.ProductIOPlugInManager.<init>(ProductIOPlugInManager.java:209)
	at org.esa.snap.core.dataio.ProductIOPlugInManager$Holder.<clinit>(ProductIOPlugInManager.java:214)
	at org.esa.snap.core.dataio.ProductIOPlugInManager.getInstance(ProductIOPlugInManager.java:45)
	at org.esa.snap.core.gpf.main.CommandLineArgs.detectWriterFormat(CommandLineArgs.java:326)
	at org.esa.snap.core.gpf.main.CommandLineArgs.parseArgs(CommandLineArgs.java:191)
	at org.esa.snap.core.gpf.main.CommandLineArgs.parseArgs(CommandLineArgs.java:79)
	at org.esa.snap.core.gpf.main.CommandLineTool.run(CommandLineTool.java:112)
	at org.esa.snap.core.gpf.main.GPT.run(GPT.java:54)
	at org.esa.snap.test.gpt.RunGPTProductReaderTest.runGPT(RunGPTProductReaderTest.java:355)

Reply:
This should actually not happen. If HDF5 cannot be initialised the exceptions are handled and only a wrning should be logged. The programm should go on.
https://github.com/senbox-org/snap-engine/blob/4873a2b1a137f2ae3571b5be318c139d9f33e2bb/snap-hdf5-writer/src/main/java/org/esa/snap/dataio/hdf5/Hdf5ProductWriterPlugIn.java#L153

As long HDF5 is not needed execution should go on as normal.

Reply:
You are right Marco that the HDF5 exception is handled and it it not thrown outside, but the stack trace is printed on the console. The first impression is that the test generateds an exception and start to see the stack trace and realize that the exception if from a writer plugin.
Running the GPT tests by means of this class in the 'snap-reader-tests' projects, no HDF5 exception is generated. Therefore something is missing in the pom.xml file of the 'snap-gpt-tests' project.
I do not known what to add (copy from pom.xml of the snap-reader-tests) in the pom.xml file to avoid the HDF5 exception.

Reply:
Actually nothing is missing in snap-gpt-tests. The snap-reader-tests don't have a dependency to snap-hdf5-writer. That's why the exception doesn't occure.

There are properties to define the path to the hdf libraries:
-Dncsa.hdf.hdflib.HDFLibrary.hdflib=c:\path\to\jhdf.dll
-Dncsa.hdf.hdf5lib.H5.hdf5lib=c:\path\to\jhdf5.dll
This libs are in lib-hdf/src/main/resources/lib
This is done for gpt tests on the server.

Probably we should update the lib-hdf, so that the libs are also found if not using the netbeans module system. But hdf is not often used anymore. So this had no high priority and we just accepted the sometimes occuring exception.

For we can leave this class where it is.
I've created an issue for the hdf-lib https://senbox.atlassian.net/browse/SNAP-1333
and mentioned also this class that it can be moved if the issue is resolved.